l10n_ch_qr_bill/models/account_invoice.py

```diff
@@ -140,7 +140,8 @@ def _space_qrr(ref):
             record.l10n_ch_qrr_spaced = spaced_qrr
 
     def _get_communications(self):
-        if self.has_qrr():
+        is_qrr = self.has_qrr() or self.partner_bank_id._is_qr_iban()
```

Comment:
I don't know all the stuff in here, but on line 330 there is a check to have both of this conditions

Reply:
To give some perspective here is the rule:

:heavy_check_mark:  QR-IBAN and QRR
:x: QR-IBAN and not QRR
:heavy_check_mark: IBAN and Free reference
:heavy_check_mark: IBAN and QRR (suspicious but considered as a Free reference thus is OK)

Which mean that we can rely solely on `self.partner_bank_id._is_qr_iban()` to know if `structured_communication` must be set or not.